### PR TITLE
including relative wind vector to FreeVario (src/Device/Driver/FreeVario.cpp)

### PR DESCRIPTION
**Pull request by Blaubart**
Including relative wind vector from external device like Larus, Anemoi, etc in FreeVario. If external wind is not available calculated wind from XCSoar/OpenSoar is displayed.

### CODE REVIEW

**Comment by lordfolken** [comment]
In the PR line
 * file path or driver name
 * Link to protocol documentation
At least https://freevario.de/en/communication-interface/
doesn't seem to list those sentences.

**Comment by Blaubart** [comment]
I added these information at https://freevario.de/en/communication-interface/ a couple hours ago:

$PFV,AWD,[value as double]
average relative wind direction of the last 30 seconds in degrees

$PFV,CWD,[value as double]
current, relative wind direction in degrees

$PFV,AWS,[value as double]
average wind speed of the last 30 seconds in km/h

$PFV,CWS,[value as double]
current wind speed in km/h

**Comment by lordfolken** [comment]
> I added these information at https://freevario.de/en/communication-interface/ a couple hours ago:
> 
> $PFV,AWD,[value as double] average relative wind direction of the last 30 seconds in degrees
> 
> $PFV,CWD,[value as double] current, relative wind direction in degrees
> 
> $PFV,AWS,[value as double] average wind speed of the last 30 seconds in m/s
> 
> $PFV,CWS,[value as double] current wind speed in m/s

Do you need sub comma precision on these sentences? Couldn't they just be an int?

**Comment by Blaubart** [comment]
Your objection is justified. But integer would also be sufficient for the speed if I converted it into km/h beforehand.

**Comment by lordfolken** [comment]
> Your objection is justified. But integer would also be sufficient for the speed if I converted it into km/h beforehand.

Sorry I meant this to be a discussion. 
Upon further research i would keep it a double. Degrees() outputs a double because XCSoar uses radians for calculations internally. Therefore you could get a float form the application.

File: src/Device/Driver/FreeVario.cpp [hunk 1]
```diff
@@ -281,6 +281,35 @@ FreeVarioDevice::OnCalculatedUpdate(const MoreData &basic,
 
   char nmeaOutbuffer[80];
 
+ if (basic.external_wind_available.IsValid() && basic.track_available){
+   int track = basic.track.Degrees();
```
Comment (lordfolken) [inline src/Device/Driver/FreeVario.cpp:285]: Save yourself the conversion to int, and do this directly in the relWindDirection below.
True for all other instances aswell.

Comment (Blaubart) [reply src/Device/Driver/FreeVario.cpp]: Sorry, then I misunderstood you. I thought that was a request from you to change that! I have now eliminated the integers and also added a query for the calculated wind as to whether the external wind is definitely invalid.

**Comment by lordfolken** [comment]
Now if you could integrate the fixup into the first commit. Then it would be perfect. 😃

**Comment by Blaubart** [comment]
OK, I'm a bit inexperienced with Github when it comes to this sort of thing, I'll give it a try ;-)

**Comment by Blaubart** [comment]
I tested the new driver on the ground today. Larus sends a new wind to XCSoar about every second, which is also updated there. But FreeVario only gets this after about 20s! Of course that is far too late. Could it be because the glider was on the ground and therefore OnCalculatedUpdate did not pass through the new wind? For example, if the altitude changes, FV displays this directly. 

It's also funny that it seems to matter which device Larus is. I tested two settings: 
- Device A -> OpenVario (Disabled)
- Device B -> Flarm
- Device C -> FreeVario
- Device D -> AirControlDisplay
- Device E -> Larus

In this configuration I get a new wind every 20s.I also tested this order:
- Device A -> OpenVario (Disabled)
- Device B -> Flarm
- Device C -> Larus
- Device D -> FreeVario
- Device E -> AirControlDisplay

This means that no wind reaches the FV at all. Does anyone have an explanation for this??

**Pull request closed without merging**